### PR TITLE
fix: adding try catch to update action

### PR DESCRIPTION
**Pull request by alex-paca-jalafund**
Fixing update action for non existent objects

### CODE REVIEW

File: LosGosus/src/ConsoleInterface/ManagersUI/BookManagerUI.cs [hunk 1]
```diff
@@ -99,15 +99,20 @@ private void UpdateBook()
...
+                AnsiConsole.MarkupLine("[green]Book updated successfully![/]");
+            }
+            else
+            {
+                ErrorHandler.HandleError(new InvalidBookException("Book is not valid. Please check the details and try again."));
+            }
+        } catch 
         {
-            Handler.HandleError(new InvalidBookException("Book is not valid. Please check the details and try again."));
+            ErrorHandler.HandleError(new InvalidBookException("Book is not valid or doesn't exists."));
```
Comment (joseluis-teran-jalafund) [inline LosGosus/src/ConsoleInterface/ManagersUI/BookManagerUI.cs]: We don't have the ErrorHandler class. Now it's called `Handler.cs`

File: LosGosus/src/ConsoleInterface/ManagersUI/PatronManagerUI.cs [hunk 2]
```diff
@@ -91,17 +91,21 @@ private void UpdatePatron()
...
-        }
-        else
+            bool success = patronController.TryUpdatePatron(name, membershipNumber, contactDetails);
+            if (success)
+            {
+                AnsiConsole.MarkupLine("[green]Patron updated successfully![/]");
+            }
+            else
+            {
+                ErrorHandler.HandleError(new InvalidPatronException("Invalid Patron information."));
```
Comment (joseluis-teran-jalafund) [inline LosGosus/src/ConsoleInterface/ManagersUI/PatronManagerUI.cs]: Same here Alex.

**Review by joseluis-teran-jalafund — CHANGES_REQUESTED** [review]
Update class instances according to the code into **release** branch.

**Comment by joseluis-teran-jalafund** [comment]
It looks good, but I'd suggest to merge first **[PR - 44](https://github.com/programacion-6/Opcion1LosGosus/pull/44)** in order to see if everything is okay. I mean build and test doesn't fail for both projects. Because it can happen what I saw while I was working on **[PR - 43](https://github.com/programacion-6/Opcion1LosGosus/pull/43)** that **release** branch was down due to code refactors were merged without updating unit tests.

**Comment by alex-paca-jalafund** [comment]
> It looks good, but I'd suggest to merge first **[PR - 44](https://github.com/programacion-6/Opcion1LosGosus/pull/44)** in order to see if everything is okay. I mean build and test doesn't fail for both projects. Because it can happen what I saw while I was working on **[PR - 43](https://github.com/programacion-6/Opcion1LosGosus/pull/43)** that **release** branch was down due to code refactors were merged without updating unit tests.

I think it will not change anything on test because its  UI method, and its a try catch in a part of a method there is nothing more

**Comment by joseluis-teran-jalafund** [comment]
> > It looks good, but I'd suggest to merge first **[PR - 44](https://github.com/programacion-6/Opcion1LosGosus/pull/44)** in order to see if everything is okay. I mean build and test doesn't fail for both projects. Because it can happen what I saw while I was working on **[PR - 43](https://github.com/programacion-6/Opcion1LosGosus/pull/43)** that **release** branch was down due to code refactors were merged without updating unit tests.
> 
> I think it will not change anything on test because its UI method, and its a try catch in a part of a method there is nothing more

build status on `LosGosus`
image

test status on `LosGosusTest`
image

Are you update with **release** branch?
image

m...  i didnt change nothing on validations on this PR but maybe we should look release branch before merging something else

**Comment by alex-paca-jalafund** [comment]
> > > It looks good, but I'd suggest to merge first **[PR - 44](https://github.com/programacion-6/Opcion1LosGosus/pull/44)** in order to see if everything is okay. I mean build and test doesn't fail for both projects. Because it can happen what I saw while I was working on **[PR - 43](https://github.com/programacion-6/Opcion1LosGosus/pull/43)** that **release** branch was down due to code refactors were merged without updating unit tests.
> > 
> > 
> > I think it will not change anything on test because its UI method, and its a try catch in a part of a method there is nothing more
> 
> build status on `LosGosus` image
> 
> test status on `LosGosusTest` image
> 
> Are you update with **release** branch? image
> 

m... i didnt change nothing on validations on this PR but maybe we should look release branch before merging something else

**Comment by joseluis-teran-jalafund** [comment]
> > > > It looks good, but I'd suggest to merge first **[PR - 44](https://github.com/programacion-6/Opcion1LosGosus/pull/44)** in order to see if everything is okay. I mean build and test doesn't fail for both projects. Because it can happen what I saw while I was working on **[PR - 43](https://github.com/programacion-6/Opcion1LosGosus/pull/43)** that **release** branch was down due to code refactors were merged without updating unit tests.
> > > 
> > > 
> > > I think it will not change anything on test because its UI method, and its a try catch in a part of a method there is nothing more
> > 
> > 
> > build status on `LosGosus` image
> > test status on `LosGosusTest` image
> > Are you update with **release** branch? image
> 
> m... i didnt change nothing on validations on this PR but maybe we should look release branch before merging something else

Yeah, I suggest you to review it. Because as I said there was code refactoring without updating unit tests and I noticed when I worked on **[PR - 43](https://github.com/programacion-6/Opcion1LosGosus/pull/43)** and pipelines of **[PR - 44](https://github.com/programacion-6/Opcion1LosGosus/pull/44)** show me the status, which was failing.

**Comment by alex-paca-jalafund** [comment]
> > > > > It looks good, but I'd suggest to merge first **[PR - 44](https://github.com/programacion-6/Opcion1LosGosus/pull/44)** in order to see if everything is okay. I mean build and test doesn't fail for both projects. Because it can happen what I saw while I was working on **[PR - 43](https://github.com/programacion-6/Opcion1LosGosus/pull/43)** that **release** branch was down due to code refactors were merged without updating unit tests.
> > > > 
> > > > 
> > > > I think it will not change anything on test because its UI method, and its a try catch in a part of a method there is nothing more
> > > 
> > > 
> > > build status on `LosGosus` image
> > > test status on `LosGosusTest` image
> > > Are you update with **release** branch? image
> > 
> > 
> > m... i didnt change nothing on validations on this PR but maybe we should look release branch before merging something else
> 
> Yeah, I suggest you to review it. Because as I said there was code refactoring without updating unit tests and I noticed when I worked on **[PR - 43](https://github.com/programacion-6/Opcion1LosGosus/pull/43)** and pipelines of **[PR - 44](https://github.com/programacion-6/Opcion1LosGosus/pull/44)** show me the status, which was failing.

yeah but that doesnt has to do nothing with this PR that its just for bug fix

**Comment by alex-paca-jalafund** [comment]
> > > > > > It looks good, but I'd suggest to merge first **[PR - 44](https://github.com/programacion-6/Opcion1LosGosus/pull/44)** in order to see if everything is okay. I mean build and test doesn't fail for both projects. Because it can happen what I saw while I was working on **[PR - 43](https://github.com/programacion-6/Opcion1LosGosus/pull/43)** that **release** branch was down due to code refactors were merged without updating unit tests.
> > > > > 
> > > > > 
> > > > > I think it will not change anything on test because its UI method, and its a try catch in a part of a method there is nothing more
> > > > 
> > > > 
> > > > build status on `LosGosus` image
> > > > test status on `LosGosusTest` image
> > > > Are you update with **release** branch? image
> > > 
> > > 
> > > m... i didnt change nothing on validations on this PR but maybe we should look release branch before merging something else
> > 
> > 
> > Yeah, I suggest you to review it. Because as I said there was code refactoring without updating unit tests and I noticed when I worked on **[PR - 43](https://github.com/programacion-6/Opcion1LosGosus/pull/43)** and pipelines of **[PR - 44](https://github.com/programacion-6/Opcion1LosGosus/pull/44)** show me the status, which was failing.
> 
> yeah but that doesnt has to do nothing with this PR that its just for bug fix

Screenshot from 2024-09-01 14-51-15

i run LosGosus and LosGosusTest directories and there is no errors, are you updated with release branch?

**Comment by joseluis-teran-jalafund** [comment]
> > > > > > > It looks good, but I'd suggest to merge first **[PR - 44](https://github.com/programacion-6/Opcion1LosGosus/pull/44)** in order to see if everything is okay. I mean build and test doesn't fail for both projects. Because it can happen what I saw while I was working on **[PR - 43](https://github.com/programacion-6/Opcion1LosGosus/pull/43)** that **release** branch was down due to code refactors were merged without updating unit tests.
> > > > > > 
> > > > > > 
> > > > > > I think it will not change anything on test because its UI method, and its a try catch in a part of a method there is nothing more
> > > > > 
> > > > > 
> > > > > build status on `LosGosus` image
> > > > > test status on `LosGosusTest` image
> > > > > Are you update with **release** branch? image
> > > > 
> > > > 
> > > > m... i didnt change nothing on validations on this PR but maybe we should look release branch before merging something else
> > > 
> > > 
> > > Yeah, I suggest you to review it. Because as I said there was code refactoring without updating unit tests and I noticed when I worked on **[PR - 43](https://github.com/programacion-6/Opcion1LosGosus/pull/43)** and pipelines of **[PR - 44](https://github.com/programacion-6/Opcion1LosGosus/pull/44)** show me the status, which was failing.
> > 
> > 
> > yeah but that doesnt has to do nothing with this PR that its just for bug fix
> 
> Screenshot from 2024-09-01 14-51-15
> 
> i run LosGosus and LosGosusTest directories and there is no errors, are you updated with release branch?

Don't you see that I was on you branch? That means you weren't update with release.

Also, now I see you made a rebase some minutes ago after you comment.

**Review by joseluis-teran-jalafund — APPROVED** [review]
Nice!

**Pull request merged**